src/connection.cpp

```diff
@@ -215,14 +215,15 @@ namespace oxen::quic
 
             cid->datalen = cidlen;
             auto* conn = static_cast<Connection*>(user_data);
+            auto& ep = conn->endpoint();
 
             if (ngtcp2_crypto_generate_stateless_reset_token(
-                        token, conn->static_secret(), NGTCP2_STATELESS_RESET_TOKENLEN, cid) != 0)
+                        token, ep._static_secret.data(), ep._static_secret.size(), cid) != 0)
                 return NGTCP2_ERR_CALLBACK_FAILURE;
 
             auto dir_str = conn->is_outbound() ? "CLIENT"s : "SERVER"s;
             log::trace(log_cat, "{} generated new CID for {}", dir_str, conn->reference_id());
-            conn->endpoint().associate_cid(cid, *conn);
+            ep.associate_cid(cid, *conn);
 
             // TODO: send new stateless reset token
```

Comment:
I think this entire PR is a great change, but I'm slightly surprised you want to include it before any of the logic to handle the stateless reset tokens is implemented?  It's pretty easy, a few functions listed in the TODO's, so if you were planning on looping that in it would be totally feasible